science-dev/Core/Inc/diagnostic_temp_sensor.h

```diff
@@ -0,0 +1,36 @@
+#pragma once
+
+#include "stm32g0xx_hal.h"
+
+#include "adc_sensor.h"
+#include <stdint.h>
+#include <math.h>
+#include <stdio.h>
+#include <string.h>
+
+#define TEMP_COEFFICIENT 19.5f // mV/Celsius
+#define ZERO_DEGREE_OUTPUT 400 //mV
+
+typedef struct {
```

Comment:
Looks legit, next time I suggest just adding in the part number in the .h file so future people can reference the datasheet.